torch_xla/csrc/aten_xla_type.cpp

```diff
@@ -145,6 +146,67 @@ void AtenInitialize() {
 
 }  // namespace
 
+namespace xla_pre_autograd {
```

Comment:
Let's move the autograd based hooks into aten_autograd_ops.cpp/h

Reply:
I tried but I feel it's a better fit here since we share a lot of helper functions like `isTrivialDilation` etc (which I think doesn't worth adding to header files). Wdyt?

Reply:
I think so.
All the users of that function is max_pool*(), which will need to go in the new files.
What about max_pool3d()?
Can you add that as well?

Reply:
I'll prefer leave that as it is for now unless we hit it in user code. We should prioritize fixing the issue upstream otherwise there're much more to add ;)

Reply:
`All the users of that function is max_pool*(), which will need to go in the new files.` 
This is not entirely true as `max_pool2d_with_indices` will still be a normal lowering and it'll stay in `aten_xla_type.cpp`. ;)

Reply:
@ailzhang Please, we keep adding will-do-it-later and rushing things in, and I am not fond of that.